lib/matplotlib/backends/web_backend/js/mpl.js

```diff
@@ -2,19 +2,53 @@
 /* global mpl */
 window.mpl = {};
 
-mpl.get_websocket_type = function () {
-    if (typeof WebSocket !== 'undefined') {
-        return WebSocket;
-    } else if (typeof MozWebSocket !== 'undefined') {
-        return MozWebSocket;
-    } else {
-        alert(
-            'Your browser does not have WebSocket support. ' +
-                'Please try Chrome, Safari or Firefox ≥ 6. ' +
-                'Firefox 4 and 5 are also supported but you ' +
-                'have to enable WebSockets in about:config.'
-        );
+class MockJsWebSocket {
+    binaryType = "blob";
+
+    constructor(fig_id) {
+        this.fig_id = fig_id;
+        this.readyState = 0;
+        this.python_web_socket = null;
+        this.python_onmessage_callback = null;
+    }
+
+    get onopen() {
+        return this._onopen;
+    }
+
+    set onopen(listener) {
+        this._onopen = listener;
+        this.readyState = 1;
+    }
+
+    open(python_onmessage_callback) {
+        this.python_onmessage_callback = python_onmessage_callback;
+        if (this.onopen != null) {
+            this.onopen();
+        }
+    }
+
+    receive_binary(content, binary=true) {
+        content = content.toJs();
+        blob = new Blob([content]);
```

Comment:
There's probably a way to do this with fewer copies.

Reply:
I found a better way using `content.getBuffer()` in 55117ee.